From 5f26f6e0f4ca8d15a24fc3dcf79b8408ed64ae5b Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Mon, 23 Jun 2014 21:42:04 -0400
Subject: [scsi] lpfc 10.2.8001.0: Removed obsolete PCI IDs from the driver

Message-id: <1403559726-20488-23-git-send-email-revers@redhat.com>
Patchwork-id: 84439
O-Subject: [RHEL7.1 e-stor PATCH 22/24] scsi: lpfc 10.2.8001.0: Removed obsolete PCI IDs from the driver.
Bugzilla: 1088574
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Dick Kennedy <rkennedy@Emulex.Com>

Submitted by James Smart James.Smart@emulex.com

Reworked per Christoph's suggestions.

--

Removed obsolete PCI IDs from the driver.

Signed-off-by: James Smart james.smart@emulex.com

http://marc.info/?l=linux-scsi&m=140067392618971&w=2
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_init.c b/drivers/scsi/lpfc/lpfc_init.c
index 04745e2..9115acd 100644
--- a/drivers/scsi/lpfc/lpfc_init.c
+++ b/drivers/scsi/lpfc/lpfc_init.c
@@ -2033,78 +2033,81 @@ lpfc_get_hba_model_desc(struct lpfc_hba *phba, uint8_t *mdp, uint8_t *descp)
 
  switch (dev_id) {
  case PCI_DEVICE_ID_FIREFLY:
-  m = (typeof(m)){"LP6000", "PCI", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LP6000", "PCI",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_SUPERFLY:
   if (vp->rev.biuRev >= 1 && vp->rev.biuRev <= 3)
-   m = (typeof(m)){"LP7000", "PCI",
-     "Fibre Channel Adapter"};
+   m = (typeof(m)){"LP7000", "PCI", ""};
   else
-   m = (typeof(m)){"LP7000E", "PCI",
-     "Fibre Channel Adapter"};
+   m = (typeof(m)){"LP7000E", "PCI", ""};
+  m.function = "Obsolete, Unsupported Fibre Channel Adapter";
   break;
  case PCI_DEVICE_ID_DRAGONFLY:
   m = (typeof(m)){"LP8000", "PCI",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_CENTAUR:
   if (FC_JEDEC_ID(vp->rev.biuRev) == CENTAUR_2G_JEDEC_ID)
-   m = (typeof(m)){"LP9002", "PCI",
-     "Fibre Channel Adapter"};
+   m = (typeof(m)){"LP9002", "PCI", ""};
   else
-   m = (typeof(m)){"LP9000", "PCI",
-     "Fibre Channel Adapter"};
+   m = (typeof(m)){"LP9000", "PCI", ""};
+  m.function = "Obsolete, Unsupported Fibre Channel Adapter";
   break;
  case PCI_DEVICE_ID_RFLY:
   m = (typeof(m)){"LP952", "PCI",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_PEGASUS:
   m = (typeof(m)){"LP9802", "PCI-X",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_THOR:
   m = (typeof(m)){"LP10000", "PCI-X",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_VIPER:
   m = (typeof(m)){"LPX1000",  "PCI-X",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_PFLY:
   m = (typeof(m)){"LP982", "PCI-X",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_TFLY:
   m = (typeof(m)){"LP1050", "PCI-X",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_HELIOS:
   m = (typeof(m)){"LP11000", "PCI-X2",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_HELIOS_SCSP:
   m = (typeof(m)){"LP11000-SP", "PCI-X2",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_HELIOS_DCSP:
   m = (typeof(m)){"LP11002-SP",  "PCI-X2",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_NEPTUNE:
-  m = (typeof(m)){"LPe1000", "PCIe", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LPe1000", "PCIe",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_NEPTUNE_SCSP:
-  m = (typeof(m)){"LPe1000-SP", "PCIe", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LPe1000-SP", "PCIe",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_NEPTUNE_DCSP:
-  m = (typeof(m)){"LPe1002-SP", "PCIe", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LPe1002-SP", "PCIe",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_BMID:
   m = (typeof(m)){"LP1150", "PCI-X2", "Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_BSMB:
-  m = (typeof(m)){"LP111", "PCI-X2", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LP111", "PCI-X2",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_ZEPHYR:
   m = (typeof(m)){"LPe11000", "PCIe", "Fibre Channel Adapter"};
@@ -2123,16 +2126,20 @@ lpfc_get_hba_model_desc(struct lpfc_hba *phba, uint8_t *mdp, uint8_t *descp)
   m = (typeof(m)){"LPe111", "PCIe", "Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_LP101:
-  m = (typeof(m)){"LP101", "PCI-X", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LP101", "PCI-X",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_LP10000S:
-  m = (typeof(m)){"LP10000-S", "PCI", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LP10000-S", "PCI",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_LP11000S:
-  m = (typeof(m)){"LP11000-S", "PCI-X2", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LP11000-S", "PCI-X2",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_LPE11000S:
-  m = (typeof(m)){"LPe11000-S", "PCIe", "Fibre Channel Adapter"};
+  m = (typeof(m)){"LPe11000-S", "PCIe",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_SAT:
   m = (typeof(m)){"LPe12000", "PCIe", "Fibre Channel Adapter"};
@@ -2153,20 +2160,21 @@ lpfc_get_hba_model_desc(struct lpfc_hba *phba, uint8_t *mdp, uint8_t *descp)
   m = (typeof(m)){"LPe12000-S", "PCIe", "Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_HORNET:
-  m = (typeof(m)){"LP21000", "PCIe", "FCoE Adapter"};
+  m = (typeof(m)){"LP21000", "PCIe",
+    "Obsolete, Unsupported FCoE Adapter"};
   GE = 1;
   break;
  case PCI_DEVICE_ID_PROTEUS_VF:
   m = (typeof(m)){"LPev12000", "PCIe IOV",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_PROTEUS_PF:
   m = (typeof(m)){"LPev12000", "PCIe IOV",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_PROTEUS_S:
   m = (typeof(m)){"LPemv12002-S", "PCIe IOV",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_TIGERSHARK:
   oneConnect = 1;
@@ -2182,17 +2190,24 @@ lpfc_get_hba_model_desc(struct lpfc_hba *phba, uint8_t *mdp, uint8_t *descp)
   break;
  case PCI_DEVICE_ID_BALIUS:
   m = (typeof(m)){"LPVe12002", "PCIe Shared I/O",
-    "Fibre Channel Adapter"};
+    "Obsolete, Unsupported Fibre Channel Adapter"};
   break;
  case PCI_DEVICE_ID_LANCER_FC:
- case PCI_DEVICE_ID_LANCER_FC_VF:
   m = (typeof(m)){"LPe16000", "PCIe", "Fibre Channel Adapter"};
   break;
+ case PCI_DEVICE_ID_LANCER_FC_VF:
+  m = (typeof(m)){"LPe16000", "PCIe",
+    "Obsolete, Unsupported Fibre Channel Adapter"};
+  break;
  case PCI_DEVICE_ID_LANCER_FCOE:
- case PCI_DEVICE_ID_LANCER_FCOE_VF:
   oneConnect = 1;
   m = (typeof(m)){"OCe15100", "PCIe", "FCoE"};
   break;
+ case PCI_DEVICE_ID_LANCER_FCOE_VF:
+  oneConnect = 1;
+  m = (typeof(m)){"OCe15100", "PCIe",
+    "Obsolete, Unsupported FCoE"};
+  break;
  case PCI_DEVICE_ID_SKYHAWK:
  case PCI_DEVICE_ID_SKYHAWK_VF:
   oneConnect = 1;
-- 
1.7.1